### PR TITLE
feat: populate fields from payment attempt in payment list

### PR DESCRIPTION
**Pull request by dracarys18**
## Type of Change

- [x] Enhancement

## Description

Populate some fields from payment attempt table in Paymentlist response.

## Motivation and Context

Populate necessary fields which we show in the dashboard from payment attempt.

## How did you test it?

Manual

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code

### CODE REVIEW

File: crates/router/src/core/payments.rs [hunk 1]
```diff
@@ -621,10 +625,23 @@ pub async fn list_payments(
...
-        .into_iter()
-        .map(types::transformers::ForeignInto::foreign_into)
-        .collect();
+    let pi = futures::stream::iter(payment_intents)
+        .filter_map(|pi| async {
+            let pa = db
+                .find_payment_attempt_by_payment_id_merchant_id(
+                    &pi.payment_id,
+                    merchant_id,
+                    merchant.storage_scheme,
```
Comment (jarnura) [inline crates/router/src/core/payments.rs:634]: Should we add a comment and say this is `olap` function call and the storage.schema is always `psql` only?

File: crates/router/src/core/payments/transformers.rs [hunk 2]
```diff
@@ -393,6 +393,30 @@ where
...
+            client_secret: pi.client_secret.map(|s| s.into()),
+            created: Some(pi.created_at),
+            currency: pi.currency.map(|c| c.to_string()).unwrap_or_default(),
+            description: pi.description,
+            metadata: pi.metadata,
+            customer_id: pi.customer_id,
+            connector: pa.connector,
+            payment_method: pa.payment_method.map(ForeignInto::foreign_into),
+            payment_method_type: pa.payment_method_type.map(ForeignInto::foreign_into),
+            ..Default::default()
```
Comment (jarnura) [inline crates/router/src/core/payments/transformers.rs:415]: we need to populate other fields also right? like `refunds`. Can you confirm with the dashboard team about this.

**Review by jarnura — CHANGES_REQUESTED** [review]

File: crates/router/src/core/payments/transformers.rs (hunk 2 above)
Comment (dracarys18) [reply crates/router/src/core/payments/transformers.rs:415]: Confirmed with dashboard team they only need `payment_method`,`connector`,`payment_method_type` fields which is being populated here.

**Review by SanchithHegde — APPROVED** [review]

**Comment by ashokkjag** [comment]
@jarnura / @dracarys18 , 

Are there limits on how many payment intents are retrieved? If payment list goes into 100s, then for each call, we'll do 100s of payment attempts lookup.

Does payment attempt always have only 1 entry for payment intent + merchant id?

File: crates/router/src/core/payments.rs [hunk 3]
```diff
@@ -609,9 +609,13 @@ pub async fn list_payments(
     merchant: storage::MerchantAccount,
     constraints: api::PaymentListConstraints,
 ) -> RouterResponse<api::PaymentListResponse> {
+    use futures::stream::StreamExt;
+
+    use crate::types::transformers::ForeignFrom;
```
Comment (jarnura) [inline crates/router/src/core/payments.rs:614]: move this import to top of the file

**Review by jarnura — CHANGES_REQUESTED** [review]
Other than import changes looks good to be

File: crates/router/src/core/payments.rs (hunk 3 above)
Comment (jarnura) [reply crates/router/src/core/payments.rs:614]: Because of feature enabled we can't move to top of the file

**Review by jarnura — APPROVED** [review]

**Comment by dracarys18** [comment]
> @jarnura / @dracarys18 ,
> 
> Are there limits on how many payment intents are retrieved? If payment list goes into 100s, then for each call, we'll do 100s of payment attempts lookup.
> 
> Does payment attempt always have only 1 entry for payment intent + merchant id?

The limit is 100.  Since the call happens to OLAP database I think these calls is feasible and not heavy to the database.

**Pull request merged**